Add overview docs (#553)
MicrosoftDocs/quantum-docs-pr: articles/overview/how-to-learn-quantum-computing.md

---
title: How to learn quantum computing
description: 
author: natke
ms.author:  nakersha
ms.date: 10/16/2019
ms.topic: article
uid: microsoft.quantum.overview.learn
---

# How to learn quantum computing

Learn how to write your first quantum program.

## Getting the basics: what do you need to know?

Fortunately, you don’t need to know quantum physics to start writing quantum applications. However, there are some concepts that are especially important to understand quantum computing. But don’t worry if you don’t know something! We will give you the resources to learn the necessary tools in a light and interactive way.

> * Complex numbers: quantum mechanics makes use of complex numbers to express quantum states. They are crucial to understand the basics of quantum computation. Here you can find a tutorial to get everything you need to know about complex numbers.

Reviewer: I'd suggest swapping this with the next line, given offline discussion of relative importances between the two.

> * Linear algebra (vector and matrices): quantum states are vectors in a vector space over the complex numbers. Quantum gates are just linear operations acting on these states. Here you can find a tutorial to get everything you need to know about linear algebra. (Link to a Binder for the tutorial on linear algebra).

Reviewer: It needs a lot of improvement, to be sure, but would it be helpful to link to the existing linear algebra concepts page (https://docs.microsoft.com/quantum/concepts/vectors-and-matrices)?

* Complex numbers: quantum mechanics makes use of complex numbers to express quantum states. They are crucial to understand the basics of quantum computation. Here you can find a tutorial to get everything you need to know about complex numbers.

> * Basic quantum mechanics: ok, I know we just said that you don’t need to know quantum physics to start coding (and it’s true!). But some basic concepts of quantum mechanics and its mathematical notation will be helpful to understand quantum programming. Here you can find a tutorial to get everything you need to know about quantum mechanics. (Link to a Binder for the tutorial on basic quantum mechanics).

Reviewer 2: Complex arithmetic tutorial is up at https://mybinder.org/v2/gh/Microsoft/QuantumKatas/master?filepath=tutorials/ComplexArithmetic/ComplexArithmetic.ipynb. The others are still in review, should be up some time this week - I'll let you know.

By the way, I would swap complex numbers and linear algebra in the text, since complex numbers typically come first in the learning path.

> * Basic quantum mechanics: ok, I know we just said that you don’t need to know quantum physics to start coding (and it’s true!). But some basic concepts of quantum mechanics and its mathematical notation will be helpful to understand quantum programming. Here you can find a tutorial to get everything you need to know about quantum mechanics. (Link to a Binder for the tutorial on basic quantum mechanics).

Reviewer: There's a number of different learning paths that readers could follow; I don't think complex numbers come first in all or even most of them?

> * Basic quantum mechanics: ok, I know we just said that you don’t need to know quantum physics to start coding (and it’s true!). But some basic concepts of quantum mechanics and its mathematical notation will be helpful to understand quantum programming. Here you can find a tutorial to get everything you need to know about quantum mechanics. (Link to a Binder for the tutorial on basic quantum mechanics).

Reviewer: Similarly here, there's a few concepts guides that already exist. They need work, but it may be helpful to link to them nonetheless.

> * Basic quantum mechanics: ok, I know we just said that you don’t need to know quantum physics to start coding (and it’s true!). But some basic concepts of quantum mechanics and its mathematical notation will be helpful to understand quantum programming. Here you can find a tutorial to get everything you need to know about quantum mechanics. (Link to a Binder for the tutorial on basic quantum mechanics).

Reviewer 3: ```suggestion
* Basic quantum mechanics: okay, I know we just said that you don’t need to know quantum physics to start coding (and it’s true!). But some basic concepts of quantum mechanics and its mathematical notation will be helpful to understand quantum programming. Here you can find a tutorial to get everything you need to know about quantum mechanics. (Link to a Binder for the tutorial on basic quantum mechanics).
```

Reviewer 3: The nitpickiest of nitpicks, but I figured might as well mention it. I think _technically_ it should be either in caps "OK" or "okay" spelled out

Author: I removed the ok :)

## Code first, then learn!

> Now that we have the basics, we are ready to start writing our first quantum programs. There are many options to get started. Here we recommend you some of them:

Reviewer: ```suggestion
Now that you have the basics, you're ready to start writing your first quantum programs. There are many options to get started. Here we recommend you some of them:
```

* [Write your first quantum program](xref:microsoft.quantum.write-program)
* Learn with the [Quantum Katas](https://github.com/microsoft/QuantumKatas): an open-source project containing programming exercises aimed at teaching quantum computing. In particular, the online version of the Quantum Katas in form of Q# Jupyter Notebooks is especially easy to use. In less than 5 min you can start to write your first quantum program without needing to install anything.

> * [Tutorials and samples]()

Reviewer: Do you want this to link to the samples repo or the samples browser (once we get everything on-boarded OK)?

Reviewer: Bumping this, now that said samples are onboarded at https://docs.microsoft.com/samples/browse/?languages=qsharp

## Dive into the theory

Maybe you want to take a deeper look into the theory of quantum mechanics and quantum computing. Here you have a list of useful material:

> * Start with our guide to [Quantum computing concepts](xref:microsoft.quantum.concepts.intro), a compilation of basic concepts for quantum computing.

Reviewer 3: ```suggestion
* Start with our guide to [quantum computing concepts](xref:microsoft.quantum.concepts.intro), a compilation of basic concepts for quantum computing.
```

Author: Fixed

* _Learn Quantum Computing with Python and Q#_ (Sarah C. Kaiser and Christopher E. Granade) provides an excellent introduction for people who have little to no experience with quantum mechanics, but some programming background.
* _Quantum Computation and Quantum Information_ (Michael A. Nielsen, Isaac L. Chuang) is the most cited text in the field of quantum computation. It is regarded as the standard text on the subject. The book assumes minimal prior experience with quantum mechanics and computer science. It is an excellent choice for those readers who want a rigorous introduction to the topic as well as for those who are looking for references for advanced concepts.
* MIT OpenCourseWare has an excellent online course imparted by Allan Adams for learning the basics of quantum mechanics. Perfect for those who want a better understanding of the underlying physics.

## Join the quantum community

> You don’t have to learn this alone, there is a big community of amateurs and experts who are willing to help you. Don’t be afraid to ask!

Reviewer: ```suggestion
You don’t have to learn this alone, there is a big community of amateurs and experts alike who are willing to help you. Don’t be afraid to ask!
```

* If you have any questions about Q# or quantum computing don’t hesitate and take a look at the Quantum Computing StackExchange site. If you don’t find your specific question you can always ask a new one. 
* Check out Q# blog and Microsoft Quantum Blog to stay up to date with the latest news and resources about Q#.

> * Check Q# Community and Awesome Q# to look for more resources and material.

Reviewer: Making these into links may be helpful, perhaps?
